Fix for #504 - surface correct validation error messages if save…

theirishpenguin opened the pull request:
... fails in an after_save block.

Note: Need to add tests, if this is deemed to be in the right direction.

lgebhardt commented:
I'm ok with this change even though it's a non-normal situation. Could you please rebase against master and add a test for the new code (check for 100% coverage using `COVERAGE=true rake test` please).

theirishpenguin commented:
Great @lgebhardt. I hope to get some time this week to complete this and will do the actions listed in your comment. Thanks for the quick reply.

theirishpenguin commented:
Hi @lgebhardt, I have rebased and updated this PR with a test.

I'm not familiar with 100% of the jsonapi-resource codebase so I may not have implemented the test in the correct style - is it okay?

lgebhardt commented on test/unit/resource/resource_test.rb:
```diff
@@ -8,6 +8,25 @@ def self.records(options)
+
+  def self.records(options)
+    options[:context].posts
+  end
```
This method isn't being called in the tests. Could you please remove it?

lgebhardt commented:
Other than the one uncovered line things look good. Thanks.

theirishpenguin commented:
Cool. Done, thanks.

lgebhardt commented:
Thanks!

theirishpenguin commented:
Great, thanks!
